front/src/applications/operationalStudies/components/ManageTrainSchedule/PowerRestrictionSelector.tsx

```diff
@@ -27,113 +27,111 @@ export default function PowerRestrictionSelector() {
   const rollingStockID: number | undefined = useSelector(getRollingStockID);
   const pathFindingID = useSelector(getPathfindingID);
   const powerRestriction = useSelector(getPowerRestriction);
-  const [powerRestrictions, setPowerRestrictions] = useState<selectorOption[] | undefined>();
-  const [getRollingStockById] = osrdEditoastApi.endpoints.getLightRollingStockById.useLazyQuery({});
-  const [getPathFindingById] = osrdMiddlewareApi.endpoints.getPathfindingById.useLazyQuery({});
-  const [getPathFindingByIdCatenaries] =
-    osrdEditoastApi.useLazyGetPathfindingByPathIdCatenariesQuery();
-  const [getRollingStockWithCurves] = osrdEditoastApi.useLazyGetRollingStockByIdQuery();
-
-  const getPowerRestrictionsList = async (rollingStockInfo: LightRollingStock) => {
-    if (rollingStockInfo.power_restrictions) {
-      const powerRestrictionsList = Object.entries(rollingStockInfo.power_restrictions).map(
-        ([key, _]: [string | undefined, string]) => ({ key, value: key })
-      );
-      if (powerRestrictionsList.length > 0) {
-        powerRestrictionsList.unshift({ key: undefined, value: t('noPowerRestriction') });
-      }
-      setPowerRestrictions(powerRestrictionsList);
-    }
+  const [powerRestrictions, setPowerRestrictions] = useState<selectorOption[]>([]);
+
+  const { data: pathFinding } = osrdMiddlewareApi.useGetPathfindingByIdQuery(
+    { id: pathFindingID as number },
+    { skip: !pathFindingID }
+  );
+
+  const { data: pathWithCatenaries } = osrdEditoastApi.useGetPathfindingByPathIdCatenariesQuery(
+    { pathId: pathFindingID as number },
+    { skip: !pathFindingID }
+  );
+
+  const { data: rollingStock, isSuccess } = osrdEditoastApi.useGetRollingStockByIdQuery(
+    { id: rollingStockID as number },
+    { skip: !rollingStockID }
+  );
+
+  const getPowerRestrictionsList = async (rollingStockInfo: RollingStock) => {
```

Comment:
```suggestion
  const getPowerRestrictionsList = (rollingStockInfo: RollingStock) => {
```
Last comment, I promise ! Also, do you think you can use useMemo rather this function with a useEffect ? :)

Reply:
Done :)